### PR TITLE
ddns-scripts: Rework DigitalOcean update script

### PR DESCRIPTION
**Pull request by giannoug**
Signed-off-by: George Giannou <giannoug@gmail.com>

Maintainer: @feckert
Compile tested: N/A
Run tested: ASUS RT-AC58U / ARMv7 Processor rev 5 (v7l) / OpenWrt 21.02.0-rc3 r16172-2aba3e9784 / LuCI openwrt-21.02 branch git-21.163.64918-ba57ec5. Tested that it updates hostnames after IP renewal

Description:
As described in #16162, the DigitalOcean update script needed some work. I changed the way it handles API updates, but I have only tested successfully renewing hostnames (which works).

I'd like to check other paths too, e.g. by removing parameters and using wrong API keys, but I can't find where `write_log` is shown 🤔

### CODE REVIEW

File: net/ddns-scripts/files/usr/lib/ddns/update_digitalocean_com_v2.sh [hunk 1]
```diff
@@ -1,51 +1,42 @@
...
+[ -z "$param_opt" ] && write_log 14 "Service section not configured correctly! Missing API domain record ID as 'Optional Parameter'"
+
+# Construct JSON payload
+json_init
+json_add_object
+	json_add_string name "$username"
+	json_add_string data "$__IP"
+json_close_object
+
+__STATUS=$(curl -Ss -X PUT "https://api.digitalocean.com/v2/domains/$domain/records/$param_opt" \
```
Comment (feckert) [inline net/ddns-scripts/files/usr/lib/ddns/update_digitalocean_com_v2.sh:30]: Could you please replace `$domain` with `${domain}` and `$param_opt` with `${param_opt}`

**Review by feckert — COMMENTED** [review]
Also please bump PKG_RELEASE number

**Comment by giannoug** [comment]
@feckert Can you tell me where can I find `write_log` messages? I don't see them in syslog or DDNS logs 🤔 I would like to test this a bit more

**Comment by feckert** [comment]
You can also write output to a log file see.
https://github.com/openwrt/packages/blob/036079b308a6e1ba1ac690a1643264696780ae5e/net/ddns-scripts/files/usr/lib/ddns/dynamic_dns_functions.sh#L274

Unfortunately, I don't have an account, so I can't test it. Please let me know when you are done with your tests. Otherwise the commit looks good from my point of view

**Comment by giannoug** [comment]
@feckert I did some tests and it looks good. Here's a log example of a wrong token:

```
Tue Nov 16 17:43:38 2021 user.warn ddns-scripts[31178]: myddns_ipv4: Curl failed: 401 digitalocean.com answered: {"id": "unauthorized", "message": "Unable to authenticate you." } - TERMINATE
Tue Nov 16 17:43:38 2021 user.warn ddns-scripts[31178]: myddns_ipv4: PID '31178' exit WITH ERROR '1' at 2021-11-16 17:43
```

Looks good :)

**Comment by feckert** [comment]
Thanks merged!

**Pull request merged**